actors/builtin/init/init_test.go

```diff
@@ -1 +1,154 @@
 package init_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/ipfs/go-cid"
+	assert "github.com/stretchr/testify/assert"
+
+	"github.com/filecoin-project/specs-actors/actors/abi"
+	"github.com/filecoin-project/specs-actors/actors/abi/big"
+	builtin "github.com/filecoin-project/specs-actors/actors/builtin"
+	_init "github.com/filecoin-project/specs-actors/actors/builtin/init"
+	"github.com/filecoin-project/specs-actors/actors/runtime"
+	"github.com/filecoin-project/specs-actors/actors/runtime/exitcode"
+	adt "github.com/filecoin-project/specs-actors/actors/util/adt"
+	mock "github.com/filecoin-project/specs-actors/support/mock"
+	tutil "github.com/filecoin-project/specs-actors/support/testing"
+)
+
+func TestConstructor(t *testing.T) {
+	actor := initHarness{_init.InitActor{}, t}
+
+	receiver := tutil.NewIDAddr(t, 1000)
+	builder := mock.NewBuilder(context.Background(), receiver).WithCaller(builtin.SystemActorAddr, builtin.SystemActorCodeID)
+	rt := builder.Build(t)
+	actor.constructAndVerify(rt)
+}
+
+func TestExec(t *testing.T) {
+	actor := initHarness{_init.InitActor{}, t}
+
+	receiver := tutil.NewIDAddr(t, 1000)
+	anne := tutil.NewIDAddr(t, 1001)
+	builder := mock.NewBuilder(context.Background(), receiver).WithCaller(builtin.SystemActorAddr, builtin.SystemActorCodeID)
+
+	t.Run("abort actors that cannot call exec", func(t *testing.T) {
+		rt := builder.Build(t)
+		actor.constructAndVerify(rt)
+
+		rt.SetCaller(anne, builtin.AccountActorCodeID)
+		rt.SetActorCodeCID(anne, builtin.AccountActorCodeID)
+		// TODO test case where `ok` == false, current impl panics
+		rt.ExpectAbort(exitcode.ErrForbidden, func() {
+			actor.execAndVerify(rt, builtin.AccountActorCodeID, []byte{})
+		})
+		rt.ExpectAbort(exitcode.ErrForbidden, func() {
+			actor.execAndVerify(rt, cid.Undef, []byte{})
+		})
+	})
+
+	var fakeParams = runtime.CBORBytes([]byte{'D', 'E', 'A', 'D', 'B', 'E', 'E', 'F'})
+	var balance = abi.NewTokenAmount(100)
+
+	t.Run("happy path exec create payment channel", func(t *testing.T) {
+		rt := builder.Build(t)
+
+		actor.constructAndVerify(rt)
+
+		// anne the account actor with 100 FIL in their balance is going to create a payment channel actor with 100 FIL.
```

Comment:
```suggestion
		// anne execs a payment channel actor with 100 FIL.
```
Reference to anne's balance is confusing here.